src/data/dfinsupp/prod.lean

```diff
@@ -0,0 +1,137 @@
+import data.dfinsupp.basic
+import tactic
+
+universes u v w
+
+namespace dfinsupp.prod
+
+variables {ι₁ : Type u} {ι₂ : Type v}
+variables {β : ι₁ → ι₂ → Type w} [Π i j, has_zero (β i j)]
+
+-- this definition is complete, and looks fairly readable - although there may be a shorter way
+def curry (f : Π₀ (ij : ι₁ × ι₂), β ij.1 ij.2) : Π₀ i, Π₀ j, β i j :=
+begin
+  refine quotient.lift_on f (λ fij, _) (λ a b H, _),
+  exact ⟦{
+    to_fun := λ i, ⟦ {
+      to_fun := λ j, fij.to_fun (i, j),
+      pre_support := fij.pre_support.map prod.snd,
```

Comment:
well this doesn't transfer easily when replacing ι₁ × ι₂ by a sigma type

Reply:
I had assumed it would be fine, but the comment you made in the other PR looked convincing